internal/server/singleprocess/service_logs.go

```diff
@@ -1,38 +1,203 @@
 package singleprocess
 
 import (
+	"context"
 	"sync"
 
 	"github.com/hashicorp/go-hclog"
 	"github.com/hashicorp/go-memdb"
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/status"
 
+	"github.com/hashicorp/waypoint/internal/server"
 	pb "github.com/hashicorp/waypoint/internal/server/gen"
+	"github.com/hashicorp/waypoint/internal/server/grpcmetadata"
+	"github.com/hashicorp/waypoint/internal/server/logbuffer"
 	"github.com/hashicorp/waypoint/internal/server/singleprocess/state"
 )
 
 // defaultLogLimitBacklog is the default backlog amount to send down.
 const defaultLogLimitBacklog = 100
 
+func (s *service) spawnLogPlugin(
```

Comment:
I think we have some decent tests around log streams. It'd be great to add tests for this use case too. I'm not sure how easy that is actually but I'd love to talk through it given it is important behavior and also because I don't want to negatively impact our existing log style.

Reply:
Ack! Got too eager on the first day back after the weekend. Meant to write this, doing so now.